### PR TITLE
Toggle enable status of menu items

### LINKED ISSUE #4795: Options should be inactive or show Error or Information Box
JabRef version 5.0

I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists-

(1) Open JabRef 5.0 and go to Tools option.
(2) Click on the link 'Push entries to external application(TeXstudio)' which comes as active.
(3) Nothing happens.

Actual behavior: Nothing happens on clicking 'Push entries to external application(TeXstudio)' option.

Workaround: open JabRef 5.0, 'Push entries to external application(TeXstudio)' option shouldn't be active or should show the error or information box like it shows when we add a new Library with no entry which results in 'This operation requires one or more entries to be selected'.

Why it matters: It's a common U.I. bug that needs to be fixed. In this P.R. I will suggest more options that could be inactive or show error or information box.

### PR DESCRIPTION
**Pull request by tobiasdiez**
Fixes #4795. Now, certain actions in the menu and toolbar are only enabled if a database is open and/or entries are selected.
Also moved remaining classes from `gui.actions` to more specific packages.

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/gui/edit/ManageKeywordsAction.java [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
-    private final JabRefFrame frame;
+    private final StateManager stateManager;
+
+    public ManageKeywordsAction(StateManager stateManager) {
+        this.stateManager = stateManager;
 
-    public ManageKeywordsAction(JabRefFrame frame) {
-        this.frame = frame;
+        this.executable.bind(needsDatabase(stateManager).and(needsEntriesSelected(stateManager)));
+        this.statusMessage.bind(BindingsHelper.ifThenElse(this.executable, "", Localization.lang("Select at least one entry to manage keywords.")));
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/edit/ManageKeywordsAction.java]: How can you bind it if it's read only?

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/edit/ManageKeywordsAction.java]: The wrapper itself is not readonly, but allows the construction of an readonly property. It's a bit confusing.

**Comment by tobiasdiez** [comment]
This is now ready for review. Sorry for the many changes, but I had to refactor a few things to make them work with the new action framework.

File: src/main/java/org/jabref/gui/push/PushToApplicationAction.java [hunk 2]
```diff
@@ -71,66 +62,50 @@ public String getDescription() {
...
-                                                            Localization.lang("This operation requires one or more entries to be selected."));
-
-            return;
+    private static String getKeyString(List<BibEntry> entries) {
+        StringBuilder result = new StringBuilder();
+        Optional<String> citeKey;
+        boolean first = true;
+        for (BibEntry bes : entries) {
+            citeKey = bes.getCiteKeyOptional();
+            if (!(citeKey.isPresent()) || citeKey.get().isEmpty()) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/push/PushToApplicationAction.java]: There is a method in our OptionalUtil, presendAndNotEmpty or simliar

File: src/main/java/org/jabref/gui/util/BindingsHelper.java [hunk 3]
```diff
@@ -187,6 +188,16 @@ protected String computeValue() {
         return new UiThreadList<>(list);
     }
 
+    public static <T> ObservableValue<T> ifThenElse(ObservableValue<Boolean> condition, T value, T other) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/util/BindingsHelper.java]: Either github does somehow have a problem with this, or it is somehow commented out`, as it's not syntax highlighted

**Comment by Siedlerchr** [comment]
Please fix the checkstyle stuff and the l10n issue

**Pull request merged**